File: sdk/v1.0.47/javascript/health.js
import { handleResponse } from './errors.js';

export class Health {
  constructor(ctx) { this._ctx = ctx; }

  async healthCheck() {
    const resp = await fetch(this._ctx.baseUrl + `/api/v1/health/`, { method: 'GET' });
    return handleResponse(resp);
  }

  async livenessCheck() {
    const resp = await fetch(this._ctx.baseUrl + `/api/v1/health/live/`, { method: 'GET' });
    return handleResponse(resp);
  }

  async readinessCheck() {
    const resp = await fetch(this._ctx.baseUrl + `/api/v1/health/ready`, { method: 'GET' });
    return handleResponse(resp);
  }
}

File: sdk/v1.0.58/python/mediaviz_sdk/health.py
from __future__ import annotations
from typing import Any
from urllib.parse import quote, urlencode
import httpx

from .errors import handle_response


class Health:
    def __init__(self, ctx) -> None:
        self._ctx = ctx

    def health_check(self) -> dict[str, Any]:
        path = '/api/v1/health/'
        with httpx.Client() as _client:
            _resp = _client.request('GET', self._ctx.base_url + path)
        return handle_response(_resp.text, _resp.status_code, dict(_resp.headers))

    def liveness_check(self) -> dict[str, Any]:
        path = '/api/v1/health/live/'
        with httpx.Client() as _client:
            _resp = _client.request('GET', self._ctx.base_url + path)
        return handle_response(_resp.text, _resp.status_code, dict(_resp.headers))

    def readiness_check(self) -> dict[str, Any]:
        path = '/api/v1/health/ready'
        with httpx.Client() as _client:
            _resp = _client.request('GET', self._ctx.base_url + path)
        return handle_response(_resp.text, _resp.status_code, dict(_resp.headers))
